homeassistant/components/device_tracker/tplink.py

```diff
@@ -12,35 +12,31 @@
 from datetime import timedelta
 
 import requests
+import voluptuous as vol
 
-from homeassistant.components.device_tracker import DOMAIN
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.device_tracker import DOMAIN, PLATFORM_SCHEMA
 from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
-from homeassistant.helpers import validate_config
 from homeassistant.util import Throttle
 
 # Return cached results if last scan was less then this time ago
 MIN_TIME_BETWEEN_SCANS = timedelta(seconds=5)
 
 _LOGGER = logging.getLogger(__name__)
 
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string
+})
+
 
 def get_scanner(hass, config):
     """Validate the configuration and return a TP-Link scanner."""
-    if not validate_config(config,
-                           {DOMAIN: [CONF_HOST, CONF_USERNAME, CONF_PASSWORD]},
-                           _LOGGER):
-        return None
-
-    scanner = Tplink4DeviceScanner(config[DOMAIN])
-
-    if not scanner.success_init:
+    for _ in range(0, 4):
```

Comment:
This is not the same logic...

Originally, it tried to create a `Tplink4DeviceScanner` then a `Tplink3DeviceScanner`, `Tplink2DeviceScanner`, `TplinkDeviceScanner`

You changed it to try to load `Tplink3DeviceScanner` 5 times.

So two problems: 1. you are going from 4 to 0, instead of 4 to 1 and 2. you're only trying to build out one of the 4 scanner versions.